torch/distributed/pipelining/schedules.py

```diff
@@ -36,12 +36,16 @@ class _ComputationType(Enum):
     FORWARD = 1
     BACKWARD = 2
     WEIGHT = 3
+    UNSHARD = 4
+    RESHARD = 5
```

Comment:
High level question naming-wise since the next PR also adds `SEND_F`, `RECV_F`, etc. Should these and unshard/reshard be under a different enum like `_CommunicationType`?

These aren't really related to what the stage is computing and also might be easier to read code-wise to distinguish between the two. And maybe we can refactor the pipeline schedule order's naming to be something like `PipelineScheduleEvents` which encompass the computation and communication

Reply:
> refactor the pipeline schedule order's naming to be something like PipelineScheduleEvents

this sgtm

> Should these and unshard/reshard be under a different enum like _CommunicationType

well in the next PR i initially separated them under _Action.computation_type and _Action.comm_type but I didn't like that and preferred to just have one 'action' field within action.  It should be renamed from _ComputationType to something else though.  Maybe _ActionType.

The reason i like them together instead of separate is becuase in the runtime we basically need a switch statement over the union of them, so having them come from 2 fields is a bit annoying.

I could probably also make _CommType and _CompType and have _Action.action: Union[_CommType, _CompType].  Wdyt?

Reply:
> I could probably also make _CommType and _CompType and have _Action.action: Union[_CommType, _CompType]. Wdyt?

I like this. I also think this is more applicable later when we want to perform validation on the comms, then you past in the list of the comm types. Or even later if we do automatic bubble estimation, we pass in the list of the comp types. Then the full lowered schedule is a mix of both. I think that is intuitive!